src/Blueprint/Blueprint/Cmdlets/BlueprintCmdletBase.cs

```diff
@@ -0,0 +1,382 @@
+
+                return graphRbacManagementClient;
+            }
+            set => graphRbacManagementClient = value;
+        }
+
+        /// <summary>
+        /// Authorization client
+        /// </summary>
+        private IAuthorizationManagementClient authorizationManagementClient;
+
+        public IAuthorizationManagementClient AuthorizationManagementClient
+        {
+            get
+            {
+                return authorizationManagementClient = authorizationManagementClient ?? AzureSession.Instance.ClientFactory.CreateArmClient<AuthorizationManagementClient>(DefaultProfile.DefaultContext, AzureEnvironment.Endpoint.ResourceManager);
+ 
+            }
+            set => authorizationManagementClient = value;
+        }
+
+        /// <summary>
+        /// ARM client
+        /// </summary>
+        private IResourceManagementClient resourceManagerClient;
+        public IResourceManagementClient ResourceManagerClient
+        { 
+            get
+            {
+                return resourceManagerClient = resourceManagerClient ?? new ResourceManagementClient(
+                                                   DefaultProfile.DefaultContext.Environment.GetEndpointAsUri(AzureEnvironment.Endpoint.ResourceManager),
+                                                   ClientCredentials);
+            }
+            set => resourceManagerClient = value;
+        }
+
+        #endregion Properties
+
+        #region Cmdlet Overrides
+        protected override void WriteExceptionError(Exception ex)
+        {
+            var aggEx = ex as AggregateException;
+
+            if (aggEx != null && aggEx.InnerExceptions != null)
+            {
+                foreach (var e in aggEx.Flatten().InnerExceptions)
+                {
+                    WriteExceptionError(e);
+                }
+
+                return;
+            }
+
+            throw ex;
+        }
+        #endregion Cmdlet Overrides
+
+        #region Protected Methods
+        /// <summary>
+        /// Check if an assignment with the same name exists
+        /// </summary>
+        /// <param name="scope"> Subscription scope</param>
+        /// <param name="name"> Assignment name</param>
+        /// <param name="invertCheck"> Inverts the check logic from check for exists to check for not-exits </param>
+        protected void CheckIfAssignmentExist(string scope, string name, bool invertCheck)
+        {
+            PSBlueprintAssignment existingAssignment = null;
+
+            try
+            {
+                existingAssignment = BlueprintClient.GetBlueprintAssignment(scope, name);
+            }
+            catch (Exception ex)
+            {
+                if (ex is CloudException cex && cex.Response.StatusCode != System.Net.HttpStatusCode.NotFound)
+                {
+                    throw new CloudException(cex.Message);
+                }
```

Comment:
This eats the exception if it's not a CloudException at which point the check is really inconclusive but following code will assume it doesn't exist. I'd probably also rename the method to ThrowIfAssignmentExists to make it clearer to the caller what will happen.

Also is CloudException the right one to throw. My expectation would be to have CloudException be fully populated with Http status, code, message, etc.

Reply:
Refactored the function and fixed the throw statement.

---
In reply to: [257093029](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257093029) [](ancestors = 257093029)